src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java

```diff
@@ -386,7 +402,30 @@ public List<BuiltCondition> getAll() {
                           Predicate.CONTAINS,
                           new JsonTerm(getHashValue(new DocValueHasher(), getPath(), v))))
               .collect(Collectors.toList());
-
+        case NIN:
+          if (values.isEmpty()) return List.of();
+          if (!this.getPath().equals(DOC_ID)) {
+            return values.stream()
+                .map(
+                    v ->
+                        BuiltCondition.of(
+                            DATA_CONTAINS,
+                            Predicate.NOT_CONTAINS,
+                            new JsonTerm(getHashValue(new DocValueHasher(), getPath(), v))))
+                .collect(Collectors.toList());
+          } else {
+            //            List<DocumentId> documentIdList = (List<DocumentId>) values;
+            return values.stream()
+                .map(
+                    v ->
+                        BuiltCondition.of(
+                            BuiltCondition.LHS.mapAccess(QUERY_TEXT_MAP_COLUMN_NAME, Values.NULL),
```

Comment:
_id can be number too. The column to use needs to be based on document id's value type.